src/background/storage.ts

```diff
@@ -30,6 +30,10 @@ export class Storage {
           urlsDeleted: 0,
         },
       },
+      isolation: {
+        // active: true,
```

Comment:
> I think there's a good argument to move the isolation.active out of preferences. My thought is that it's not really a preference as it cannot be toggled from the Preferences / Options page. It's more of a pseudo-pref that's only in the popup. I think I can cleanly pull it out of prefs and into the new local storage I added to resume the auto-re-isolate. That might also resolve #433.

Yeah, I think that's a good point. I'm not sure why I've added it to the preferences in the first place, but I think it was because I intended to allowed disabled isolation to be part of the preferences. Now that I think about it I'm not sure whether it's a good idea, hm.

I was even thinking about making the automatic reactivation delay a default setting - feels like that would make sense, with something like 5 minutes as default? In that case it'd make even less sense to make `active` part of the preferences I think.

What do you think?

Reply:
> Yeah, I think that's a good point. I'm not sure why I've added it to the preferences in the first place, but I think it was because I intended to allow disabled isolation to be part of the preferences. Now that I think about it I'm not sure whether it's a good idea, hm.

Your call on this one.  The #433 bug is pretty obscure.  It's going to touch a lot of files.  I might suggest detaching it from this PR - although they're so similar if I move forward, I'll probably branch my branch and keep the code dependent while using a separate PR.

> I was even thinking about making the automatic reactivation delay a default setting - feels like that would make sense, with something like 5 minutes as default? In that case it'd make even less sense to make `active` part of the preferences I think.
> What do you think?

I thought a similar default...  but for "beginner" users having the browser do something automagically in the background could be confusing.  That said...  I'll bet there aren't many beginners using this extension.  My instinct says leave it off by default for some reason still.

Reply:
>  I might suggest detaching it from this PR

Yeah, that sounds reasonable. Would be nice to remove the `// active: true` then. My feeling is that exporting preferences with isolation toggled off shouldn't be something that's possible. So refactoring it into storage (and removing it for existing users from the preferences in `m̀igration.ts`) would resolve #433.  If you're interested in working on this as a follow-up at some point, that'd be nice - but of course don't feel obliged to.

>  My instinct says leave it off by default for some reason still.

You're probably right. Lets leave it off for now, can always change it later on with some more experience on how the feature feels.

Reply:
Extra comments removed on current commit.

If you're willing to put up with me, I can take a crack at the refactoring.  I'm glad you mentioned migration.ts, though...  I probably would have missed that.  I probably won't be able to dive in until next weekend.